src/store/models/modals.ts

```diff
@@ -235,6 +253,102 @@ const modalsModel: ModalsModel = {
       defaultCommand: undefined,
     });
   }),
+  setBalanceChannels: action((state, payload) => {
+    state.balanceChannels = {
+      ...state.balanceChannels,
+      ...payload,
+    };
+  }),
+  showBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: true });
+  }),
+  hideBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: false });
+  }),
```

Comment:
Sorry, but it looks like you moved too much of the code into this store. Only these 3 actions should live here to control whether the modal is visible and to pass along any initial state needed. 

The other actions probably live in the `lightning` store since these deal with channels and invoices.

Reply:
I think the ideal place would be in `BalanceChannelsModal` since this data only needs to exist while the modal is on the screen. Personally, I would have implemented this slightly differently where the values you have in `channelsInfo` would be form inputs in the component. You could use hidden fields for values that do not need to be visible to the user, like the channel id and node names. Then when the form is submitted, you'll have the new balances from the sliders along with all of the other values from the hidden fields.